### PR TITLE
fix(api): fix migration issues found during testing

### PR DESCRIPTION
**Pull request by gustavosbarreto**
## Summary

Tested the MongoDB-to-PostgreSQL migration against multiple databases. All pass with 0 mismatches and 0 missing records.

**Schema fixes:**
- Widen `username` to `varchar(64)`, `mac` to `varchar(64)`, session `id` to `varchar(128)`
- Replace `session_event_type` enum with `varchar(64)` — real data has types not in the original enum
- Add `heredoc` and `unknown` to `session_type` enum

**Orphan handling** (records whose parent was deleted in Mongo but child remained):
- Skip devices/tags with orphaned namespaces
- Skip sessions with orphaned devices, session_events with orphaned sessions
- Skip memberships with orphaned users
- Clear dangling `preferred_namespace` references on users

**Data normalization:**
- Lowercase `role`, `origin`, `status`, `auth_methods` — some Mongo docs had mixed case
- Derive username from email for SSO users with NULL username

**Validation:**
- Count validation now treats PG < Mongo as a warning (orphans skipped), not a fatal error
- Add `scope: "core"` to all migration log entries for filtering

### CODE REVIEW

File: api/store/migrate/users.go [hunk 1]
```diff
@@ -40,18 +42,36 @@ type mongoUserPrefs struct {
...
+	if username != "" {
+		return username
+	}
+
+	if email != "" {
+		return email
+	}
+
+	return "unknown"
+}
```
[the remark this answers is not in the document]
Comment (gustavosbarreto) [reply api/store/migrate/users.go:57]: Won't address: the disambiguation logic at lines 140-152 already handles duplicate usernames by appending a suffix. Using the local part keeps usernames readable; full email as fallback is intentional for the rare case where the local part is empty.

File: api/store/migrate/keys.go [hunk 2]
```diff
@@ -138,7 +154,7 @@ func (m *Migrator) migratePublicKeyTags(ctx context.Context) error {
 		total += len(batch)
 	}
 
-	log.WithField("count", total).Info("Migrated public_key_tags")
+	log.WithFields(log.Fields{"scope": "core", "count": total}).Info("Migrated public_key_tags")
```
[the remark this answers is not in the document]
Comment (gustavosbarreto) [reply api/store/migrate/keys.go:166]: Addressed — added orphan filtering via `loadValidNamespaces` to `migratePublicKeyTags`.

File: api/store/migrate/users.go [hunk 3]
```diff
@@ -88,14 +115,43 @@ func (m *Migrator) migrateUsers(ctx context.Context) error {
...
+			u.Username = fmt.Sprintf("%s-%d", original, n+1)
+			usernames[original] = n + 1
+
+			log.WithFields(log.Fields{
+				"scope":    "core",
+				"original": original,
+				"resolved": u.Username,
+			}).Warn("Disambiguated duplicate username")
+		}
+		usernames[u.Username] = 0
```
[the remark this answers is not in the document]
Comment (gustavosbarreto) [reply api/store/migrate/users.go]: Won't address: edge case is unlikely with real data. The disambiguation handles the common case; if a three-way collision happens in practice we'll handle it then.

[the remark this answers is not in the document]
Comment (gustavosbarreto) [reply api/store/migrate/users.go]: Won't address: migration code doesn't need unit test coverage for this edge case right now.

[the remark this answers is not in the document]
Comment (gustavosbarreto) [reply api/store/migrate/users.go]: Won't address: removed the disambiguation logic entirely. No username collisions occurred across all 11 production databases tested.

[the remark this answers is not in the document]
Comment (gustavosbarreto) [reply api/store/migrate/users.go]: Won't address: disambiguation logic removed — not needed based on production data.

File: api/store/migrate/deep_validate_users.go [hunk 4]
```diff
@@ -84,7 +89,17 @@ func (m *Migrator) compareUserBatch(ctx context.Context, r *ValidationReport, ba
...
 		r.CheckTime(t, id, "LastLogin", exp.LastLogin, act.LastLogin)
-		r.CheckField(t, id, "PreferredNamespace", exp.Preferences.PreferredNamespace, act.Preferences.PreferredNamespace)
+
+		// PreferredNamespace may have been cleared during migration if it
+		// pointed to a namespace that no longer exists.
+		expNS := exp.Preferences.PreferredNamespace
+		actNS := act.Preferences.PreferredNamespace
+		if _, nsExists := validNS[expNS]; actNS == "" && !nsExists {
+			// Cleared dangling reference — expected.
+		} else {
```
[the remark this answers is not in the document]
Comment (gustavosbarreto) [reply api/store/migrate/deep_validate_users.go:99]: Addressed: removed disambiguation logic, so this is no longer applicable.
